### ISSUE TITLE
Search Editor Progress Bar Animates Endlessly After Search Completed (intermittent)

### ISSUE THREAD

**Issue by karlhorky** [microsoft/vscode#225315]
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.92.1 Commit eaa41d57266683296de7d118f574d0c2652e1fc4
- OS Version: macOS Sonoma 14.6.1 (23G93)

Steps to Reproduce:

1. Perform a search using the [VS Code Search Editor](https://code.visualstudio.com/docs/editor/codebasics#_search-editor)
2. Intermittent: Blue progress bar / indicator animates endlessly, even after search is completed (I have not yet identified a pattern for which search terms trigger this) - almost like hanging behavior, but everything is still responsive.
3. Reloading the editor using "Developer: Reload Window" causes the problem to disappear

Blue progress bar indicator:

![Screenshot 2024-08-10 at 13 05 59](https://github.com/user-attachments/assets/fad42ad3-dee7-42d2-b5c5-732931677d5f)

https://github.com/user-attachments/assets/dcacab77-5e07-4c7a-b0d1-d0b3a2dbad51

Search pattern (invalid regular expression):

```
pageTitle: '([^']+)',\n.+)
```

Fixing the invalid regular expression problem does not make the problem go away:

https://github.com/user-attachments/assets/343dade1-9cc8-4f02-8e6a-3f2cb1abefd4

This is not a new problem - this has been happening (intermittently) since shortly after the Search Editor was introduced.

cc @andreamah @rzhao271 @roblourens @JacksonKearl

**Comment by andreamah** [comment]
To clarify:

1. Open search editor and enable regex.
2. Search for `pageTitle: '([^']+)',\n.+)`
3. You get a loading animation that can't be stopped without closing the editor

**Comment by karlhorky** [comment]
Nice, thanks for the confirmation! ❤️

Just to be certain - I'm not sure that an invalid regular expression is the only way to trigger the endless search loading indicator.

If I see it reproduced with a different pattern, I will add a new comment.

**Comment by andreamah** [comment]
From https://github.com/microsoft/vscode/pull/225549#issuecomment-2334541846

> To be clear, the infinite loading bar doesn't appear right away. If I just paste `pageTitle: '([^']+)',\n.+)` into the field, then there is no bar, only the "Invalid regular expression" error like in your screenshot.
> 
> It's when I move from a valid regex to an invalid one, repro steps:
> 
> 1. Paste in `pageTitle: '([^']+)',\n.+)` - infinite loading bar not reproduced yet
> 2. Press backspace 1 time - infinite loading bar not reproduced yet
> 3. Add back `)` - infinite loading bar reproduced
> 
> Maybe also needs a few cycles of the above steps, if the first time doesn't work.

**Comment by karlhorky** [comment]
@osortega hi, I see you're taking over this issue! Happy to provide any further context here, as you need.

Would love to get this fixed!

**Comment by osortega** [comment]
> 1. Paste in pageTitle: '([^']+)',\n.+) - infinite loading bar not reproduced yet
> 2. Press backspace 1 time - infinite loading bar not reproduced yet
> 3. Add back ) - infinite loading bar reproduced

@karlhorky I tried reproducing this issue with the steps above but I still see the red error message instead of the infinite loading bar.
Are you still having this problem? If so, what are the steps you are taking?

**Comment by karlhorky** [comment]
@osortega yes, it appears the bug has been fixed in one of the last releases, thanks for checking in to this again!

Closing for now - if I in future notice the animating endlessly / infinite loading behavior again, I'll reply here again.

**Comment by karlhorky** [comment]
@osortega just saw it again, this time with the regular expression search term `not found \|`

https://github.com/user-attachments/assets/4975689f-2d70-414f-a12e-22c21746905b

I'll see how easy it is to reproduce...

**Comment by karlhorky** [comment]
Ok, seems easy to reproduce:

1. Open a search editor
2. Enable Regular Expressions
3. Add the search term `not found ` and wait for it to load
4. Add the search term `not found \` (invalid regex) and wait for it to load
5. Optional: change back to a valid search term like `not found `
